controller/meeting/index.js

```diff
@@ -0,0 +1,247 @@
+    return docs;
+  });
+  await Ydb.disconnect();
+  trips.forEach((tour) => {
+    if (tourist.tours.includes(tour._id)
+      && tour.ending_date > Date.now()
+      && (!currentTour
+        || tour.beginning_date < currentTour.beginning_date)) {
+      currentTour = tour;
+    }
+  });
+  return currentTour;
+};
+
+const output = (obj) => {
+  const Format = require('../utils/format');
+  return `❗️ Информация про встречу:\n
+📅 Дата: ${Format.formatDate(obj.date)} \r
+🕑 Время: ${obj.time} \r
+🏛 Место: ${obj.place_name} \r
+🗺 Точный адрес: ${obj.place_address}`;
+};
+
+const showDirection = (req, send) => {
+  const start = req.body.message.location;
+  const sentMessage = req.body.message.text;
+
+  if (sentMessage !== 'Cancel operation') {
+    const options = `${start.latitude},${start.longitude}/${end.lat},${end.lng}`;
+    const link = `https://www.google.com.ua/maps/dir/${options}?hl=ru`;
+    send(`📍 Маршрут к месту встречи: \n${link}`, 'none');
+  } else {
+    send('Операция отменена.', 'none');
+  }
+  return 'WAITING COMMAND';
+};
+
+const setTime = async (req, tour, send) => {
+  const chatId = req.body.message.chat.id;
+  const sentMessage = req.body.message.text;
+
+  if (timeValidation(sentMessage)) {
+    const meetDate = new Date(tour.date.valueOf());
+    const meetingDate = new Date(meetDate.setUTCDate(meetDate.getUTCDate() + (tour.day - 1)));
+
+    const Mdb = require('../../db/meeting-bot');
+    const mconn = await Mdb.connect();
+    const Info = mconn.models.info;
+    const note = await Info.findOne({ tour_id: tour.id }, (err, docs) => {
+      if (err) return console.error(err);
+      return docs;
+    });
+    if (!note) {
+      const mongoose = require('mongoose');
+      Info.create(
+        {
+          _id: new mongoose.Types.ObjectId(),
+          tour_id: tour.id,
+          date: meetingDate,
+          time: sentMessage.replace(/\.|-/g, ':')
+        },
+        (err, doc) => {
+          if (err) return console.error(err);
+          return doc;
+        }
+      );
+    } else {
+      Info.findByIdAndUpdate(note._id,
+        {
+          date: meetingDate,
+          time: sentMessage.replace(/\.|-/g, ':')
+        },
+        (err, doc) => {
+          if (err) return console.error(err);
+          return doc;
+        });
+    }
+
+    createJob(60, meetingDate, sentMessage.replace(/\.|-/g, ':'), send);
+    createJob(30, meetingDate, sentMessage.replace(/\.|-/g, ':'), send);
+    createJob(15, meetingDate, sentMessage.replace(/\.|-/g, ':'), send);
+
+    send(chatId, 'Время успешно задано.', 'admin');
+    return 'WAITING COMMAND';
+  }
+  send(chatId, 'Время введено в некорректном формате. Пожалуйста, введите снова.', 'none');
+  return 'WAITING TIME AGAIN';
+};
+
+const createJob = (min, meetingDate, time, send) => {
+  const schedule = require('node-schedule');
+  const date = new Date(meetingDate.valueOf());
+  date.setUTCHours(Number(time.split(':')[0]) - min);
```

Comment:
use day.js or some lib instead, working with date as with strings may cause errors

Reply:
time is a variable, that contains a message, which is sent by a user. It's not a Date type